gematria/datasets/find_accessed_addrs_exegesis.cc

```diff
@@ -152,6 +152,11 @@ Expected<AccessedAddrs> ExegesisAnnotator::findAccessedAddrs(
           uintptr_t MapAddress = (CrashInfo.getAddress() & ~0xfff);
           if (MapAddress == 0)
             return make_error<Failure>("Segfault at zero address, cannot map.");
+          for (const auto &MemoryMapping : BenchCode.Key.MemoryMappings) {
+            if (MemoryMapping.Address == MapAddress)
+              return make_error<Failure>(
+                  "Segfault at an address already mapped.");
+          }
```

Comment:
What kind of error happens here? Do I understand correctly that this happens when `mmap()` fails to map the memory we asked it to map?

If yes, can we catch this earlier? E.g. during the mmaping phase?

Reply:
I haven't done too much investigation (yet), but that's what seems to be happening. I agree that exegesis needs correct error handling on `mmap` failures, I just haven't had time to implement since it's relatively complicated (requires writing code to manually generate the assembly for the error handling). It's on my TODO list, but somewhat low priority (will get bumped up with this issue).

Reply:
SG, then please add a TODO note also here.